pipelines/batch/src/main/java/com/google/fhir/analytics/JdbcFetchHapi.java

```diff
@@ -172,62 +218,108 @@ public FetchRowsJdbcIo(
 
       builder =
           new StringBuilder(
-              "select td.tag_code, td.tag_display, td.tag_system, td.tag_type, tag.res_id from"
-                  + " hfj_res_tag tag join hfj_tag_def td on tag.tag_id = td.tag_id join"
-                  + " hfj_resource res on tag.res_id = res.res_id where res.res_type = ? and"
+              "SELECT td.tag_code, td.tag_display, td.tag_system, td.tag_type, tag.res_id FROM"
+                  + " hfj_res_tag tag JOIN hfj_tag_def td ON tag.tag_id = td.tag_id JOIN"
+                  + " hfj_resource res ON tag.res_id = res.res_id WHERE res.res_type = ? AND"
                   + " res.res_id % ? = ? ");
       if (since != null && !since.isEmpty()) {
         builder.append("  and res.res_updated > '").append(since).append("'");
       }
       tagQuery = builder.toString();
       log.info("JDBC query for tags: " + tagQuery);
+
+      builder =
+          new StringBuilder(
+              "SELECT res_link.src_resource_id, golden.fhir_id AS golden_fhir_id, source.fhir_id AS"
+                  + " source_fhir_id FROM hfj_res_link res_link JOIN mpi_link mdm_link ON"
+                  + " res_link.target_resource_id = mdm_link.target_pid JOIN hfj_resource golden ON"
+                  + " golden.res_id = mdm_link.golden_resource_pid JOIN hfj_resource source ON"
+                  + " source.res_id = mdm_link.target_pid WHERE res_link.source_resource_type = ?"
+                  + " AND res_link.src_resource_id % ? = ? AND"
+                  + " mdm_link.match_result = 2 OR mdm_link.match_result = 1");
```

Comment:
Should we just make this "or" added based on the property? Or did you remove that setting?

Reply:
I'm thinking we do that in a separate PR as it complicates the other `SELECT` quite a bit.